message-index/messages/GHC-77539/index.md

```diff
@@ -0,0 +1,21 @@
+---
+title: Illegal tuple constraint
+summary: A tuple of constraints was used without enabling the ConstraintKinds extension
+extension: ConstraintKinds
+introduced: 9.6.1
+severity: error
+---
+
+Constraints are the part of a signature that defines the type classes that must be implemented for the types used to instantiate the type variables, found to the left of the double arrow (`=>`).
+In [Haskell 2010](https://www.haskell.org/onlinereport/haskell2010/haskellch4.html#x10-630004.1), type class constraints are either:
+
+ * a single constraint that consists of a named class applied to arguments, or
+
+ * multiple constraints in parentheses, separated by commas.
+
+This strict syntax is necessary because type classes do not themselves form types in Haskell 2010.
```

Comment:
how about `Doing a => Constraints b => Like c => This d => a -> ...`?

Reply:
AFAIK multiple sets of constraints are fine in both H98 and H2010, but they still have to either be `C a` or `(C1 a, C2 a, ...)`.

Reply:
In any case, they don't trigger this particular error from what I can tell